xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp

```diff
@@ -330,19 +289,38 @@ void CAddonVideoCodec::Reset()
   {
     if (ret == VIDEOCODEC_RETVAL::VC_PICTURE)
     {
-      m_bufferPool->ReleaseBuffer(m_lastPictureBuffer);
+      videoBuffer = m_map[m_lastPictureBuffer];
+      if (videoBuffer)
+        videoBuffer->Release();
+      m_map.erase(m_lastPictureBuffer);
       m_lastPictureBuffer = picture.decodedData;
     }
   }
-  m_bufferPool->ReleaseBuffer(m_lastPictureBuffer);
+  videoBuffer = m_map[m_lastPictureBuffer];
+  if (videoBuffer)
+    videoBuffer->Release();
+  m_map.erase(m_lastPictureBuffer);
   m_lastPictureBuffer = nullptr;
 
   m_struct.toAddon.reset(&m_struct);
 }
 
 bool CAddonVideoCodec::GetFrameBuffer(VIDEOCODEC_PICTURE &picture)
 {
-  return m_bufferPool->GetBuffer(picture);
+  CVideoBuffer *videoBuffer = m_videoBufferManager.Get(AV_PIX_FMT_YUV420P, picture.decodedDataSize);
```

Comment:
Will do. 
It's a bit more complicated - YUV has a 3/2 factor,
and you really need to use the aligned width/height.
ffmpeg when decoding H.264 1280x720 will actually require a buffer of 1536x722 due to alignment requirements of various bits of simd code.
But the client that uses the buffer will need to know about that.